### DISCUSSION TITLE
Get canvasX and canvasY for a point entry on the chart

### DISCUSSION THREAD

**Discussion by kuti-JP** [patrykandpatrick/vico#884] [Questions]
### Question

Thanks for the great effort put into this library. I'm looking for a way to get pixel coordinate of a point entry on the chart just like `LineCartesianLayerMarkerTarget` has canvasX and canvasY

### Vico version(s)

2.0.0-alpha.26

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment]
Hello @kuti-JP. Please describe what you are looking to achieve instead of an attempted solution. What would you like to use these coordinates for?

**Reply by kuti-JP** [reply]
Thanks for the swift response, I'm adding some custom layer of on top of the chart and basically this depends on the position of persistent markers. Unlike markers when they move, I can get the `canvasX` and `canvasY` but I looking for a way to do this with persistent markers.

**Reply by Gowsky** [reply]
Can you share a bit more details? What do you mean by "some custom layer"? Are you using the `Decoration` API?

**Reply by kuti-JP** [reply]
No not the decoration API as it doesn't work for my current use case since I have some interactions on the custom layer. I'm using a composable with `onGloballyPositioned` modifier to position this on the chart. I've been able to implement this with some maths since the points on the axis are spaced equally. 

The new issue I'm facing now is when the chart is scrollable and zoomed I'll need to update my calculations to include zoom & scroll factors so that part is a bit tricky hence why I'm asking if there's a way the chart can provide the position of a line entry on the canvas. For some more context I'm rewriting a graph using Vico and the xml graph library had this function that could return this values.

**Reply by Gowsky** [reply]
There isn't a dedicated API for this currently. Please feel free to open a feature request. In the meantime, you can easily solve this by, for example, creating a `Decoration` that doesn't draw anything but reports the position of the point of interest back to your custom layer. All of the required information is available in the `Decoration` functions. (You can also use any other callback or interface function that has a `CartesianDrawingContext` parameter and is called for every frame.)
